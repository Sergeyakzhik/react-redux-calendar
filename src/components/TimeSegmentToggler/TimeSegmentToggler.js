import React from 'react';
import './TimeSegmentToggler.css';

class TimeSegmentToggler extends React.Component {
  render() {
    return (
      <select onChange={this.props.onPeriodChange} className="form-control">
        <option>Month</option>
        <option>Week</option>
        <option>Day</option>
      </select>
    );
  }
}

export default TimeSegmentToggler;
