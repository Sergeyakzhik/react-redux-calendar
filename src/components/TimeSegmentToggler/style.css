.form-control {
  color: black;
  user-select: none;
  outline: 0;
  margin-top: 0.3rem;
  width: 7rem;
  display: inline-block;
  height: 2.7rem;
}

.form-control:focus {
  color: black;
  border-color: white;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 255, 255, 0.25);
}

.custom-radio .custom-control-input:checked~.custom-control-label::before {
  background-color: #5dd004;
}
