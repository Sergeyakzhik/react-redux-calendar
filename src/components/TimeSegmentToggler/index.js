import React from 'react';
import './style.css';

let TimeSegmentToggler = props => (
  <select onChange={props.onPeriodChange} className="form-control">
    <option>Month</option>
    <option>Week</option>
    <option>Day</option>
  </select>
);

export default TimeSegmentToggler;
