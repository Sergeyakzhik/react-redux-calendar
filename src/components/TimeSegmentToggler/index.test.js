import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

import TimeSegmentToggler from './index';

Enzyme.configure({ adapter: new Adapter() })

describe('TimeSegmentToggler', () => {
  let props = {
    onPeriodChange: () => {}
  }
  let component = shallow(<TimeSegmentToggler {...props} />);

  it('renders TimeSegmentToggler component', () => {
    expect(component).toMatchSnapshot();
  })
})
