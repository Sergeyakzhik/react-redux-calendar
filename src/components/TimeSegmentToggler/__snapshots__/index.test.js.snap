// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSegmentToggler render TimeSegmentToggler component 1`] = `
<select
  className="form-control"
  onChange={[Function]}
>
  <option>
    Month
  </option>
  <option>
    Week
  </option>
  <option>
    Day
  </option>
</select>
`;
